app/workers/users/record_field_test_event_worker.rb

```diff
@@ -0,0 +1,31 @@
+module Users
+  class RecordFieldTestEventWorker
+    include Sidekiq::Worker
+    include FieldTest::Helpers
+
+    sidekiq_options queue: :low_priority, retry: 10
+
+    def perform(user_id, experiment, goal)
+      return if user_id.blank?
+
+      user = User.find(user_id)
+
+      if goal == "makes_article_pageview_four_days_in_week"
+        determine_pageview_goal(user, experiment)
+      else
+        field_test_converted(experiment, participant: user, goal: goal)
+      end
+    end
+
+    private
+
+    def determine_pageview_goal(user, experiment)
+      past_week_page_view_counts = user.page_views.where("created_at > ?", 7.days.ago).
+        group_by_day(:created_at).count.values
+      past_week_page_view_counts.delete(0)
```

Comment:
why this?

Reply:
This is to determine the goal of "user visits the site 4/7 days this week. Basically to have a couple goals that account for immediate activity (reactions/comments) and another that accounts for whether users in this experiment come back.

First it counts that page views per day, like `[1,5,0,6,8,4,0]` and then counts the amount that aren't 0.

Reply:
Thanks! Let's add this as a comment :) We shouldn't rely on github conversations for this knowledge :)

Reply:
I think this feels roundabout to me because the goal of "visits 4 out of the last 7 days" has a rolling time window build into it. If a user visits 4 days in a row, then doesn't visit for 4 days, do they "unconvert'?

I'm not sure what a better metric is, just wanted to mention it.